Source/WTF/wtf/glib/FileSystemGlib.cpp

```diff
@@ -76,6 +76,21 @@ CString currentExecutablePath()
 {
     return { };
 }
+#elif OS(QNX)
+#include <fcntl.h>
+
+CString currentExecutablePath()
+{
+    static char readBuffer[PATH_MAX];
+    int selfFd = open("/proc/self/exefile", O_RDONLY);
+    ssize_t result = read(selfFd, readBuffer, sizeof(readBuffer));
+    close(selfFd);
+    if (result == -1)
+        return { };
+WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // QNX port
+    return CString(std::span { readBuffer, static_cast<size_t>(result) });
+WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
```

Comment:
A better way to write the same thing, without `WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN/END`:

```suggestion
    return unsafeMakeSpan(readBuffer, static_cast<size_t>(result));
```

Reply:
Um, it looks like I'm not smart enough to make the GitHub suggestion work properly. That's intended to replace all three lines, not just one.

Reply:
GitHub is a bit weird sometimes, for example that funny diff for stackbounds